src/devtools/client/webconsole/utils/autocomplete-eager.ts

```diff
@@ -0,0 +1,29 @@
+import { ThreadFront } from "protocol/thread";
+import { getPropertiesForObject } from "./autocomplete";
+
+// Use eager eval to get the properties of the last complete object in the expression.
+export async function getEvaluatedProperties(expression: string): Promise<string[]> {
+  const { asyncIndex, frameId } = gToolbox.getPanel("debugger")!.getFrameId();
+  let properties: string[] = [];
+
+  try {
+    const { returned, exception, failed } = await ThreadFront.evaluate(
+      asyncIndex,
+      frameId,
+      expression,
+      true
+    );
+    if (returned && !(failed || exception)) {
+      const evaluatedProperties = getPropertiesForObject(returned.getObject());
+      properties.push(...evaluatedProperties);
```

Comment:
We can do `return [...evaluatedProperties]` and also lose the var declaration at the top.